### DISCUSSION TITLE
Why <C-m> and <CR> are linked?

### DISCUSSION THREAD

**Discussion by Jaehaks** [neovim/neovim#30669] [Q&A] [answered]
Keymap for `<C-m>` makes change behavior of `<CR>`
I don't understand why this behavior is remained in neovim v0.10.2

**Comment by zeertzjq** [comment, marked as answer]
Many terminal emulators send the same sequence for `<C-m>` and `<CR>`, and in such terminal emulators Nvim cannot tell whether `<C-m>` or `<CR>` is used. Since Nvim supports multiple UIs possibly running in different terminal emulators, Nvim cannot make the behavior of mappings depend on the terminal emulator. As a compromise, when only one of `<C-m>` and `<CR>` are mapped, the mapping applies to both. If both `<C-m>` and `<CR>` are mapped, the mappings work separately, but that'll only actually work if the terminal emulator sends different sequences for `<C-m>` and `<CR>`.

**Reply by Jaehaks** [reply]
I misunderstand, Thanks:)
